packet/vlan.go

```diff
@@ -139,3 +139,30 @@ func (packet *Packet) RemoveVLANTag() bool {
 	}
 	return true
 }
+
+// ParseDataCheckVLAN parses L3, L4 and fills the field packet.Data.
+// returns 0 in case of success and -1 in case of
+// failure to parse L3 or L4. VLAN presence is checked.
+func (packet *Packet) ParseDataCheckVLAN() int {
+	var pktTCP *TCPHdr
+	var pktUDP *UDPHdr
+	var pktICMP *ICMPHdr
+
+	pktIPv4, pktIPv6, _ := packet.ParseAllKnownL3CheckVLAN()
```

Comment:
Why without ARP?

Reply:
ParseData() does not parse ARP too
ARP packets has no user Data

Reply:
Ok, get it.